src/xas_tdp_correction.F

```diff
@@ -0,0 +1,1870 @@
+                                        notcontract_1=[1], contract_2=[3], notcontract_2=[1, 2], &
+                                        map_1=[1], map_2=[2, 3], bounds_2=[1, nhomo], bounds_3=bounds_2d)
+
+                  CALL calc_ss_oov_contrib(parts(1), parts(2), ja_Ik, homo_evals, lumo_evals, omega_k, c_ss)
+
+               END IF !c_ss !!= 0
+
+               CALL dbcsr_t_destroy(ja_Ik)
+            END DO
+
+            DO ibatch = 1, nbatch_virt
+
+               virt_bo = get_limit(nblk_virt, nbatch_virt, ibatch - 1)
+               bounds_1d = [SUM(mo_blk_size(1:nblk_occ + virt_bo(1) - 1)) + 1, &
+                            SUM(mo_blk_size(1:nblk_occ + virt_bo(2)))]
+
+               CALL dbcsr_t_create(mo_template, aj_Ib)
+               CALL dbcsr_t_contract(alpha=dbcsr_scalar(1.0_dp), tensor_1=aj_X, tensor_2=oI_Y(ido_mo), &
+                                     beta=dbcsr_scalar(0.0_dp), tensor_3=aj_Ib, contract_1=[3], &
+                                     notcontract_1=[1, 2], contract_2=[2], notcontract_2=[1], &
+                                     map_1=[1, 2], map_2=[3], bounds_3=bounds_1d)
+
+               !opposite-spin contribution
+               CALL calc_os_ovv_contrib(parts(3), parts(4), aj_Ib, lumo_evals, homo_evals, lumo_evals, &
+                                        omega_k, c_os, nhomo, nhomo)
+
+               !same-spin contribution, only if c_ss is not 0
+               !directly compute the difference (aj|Ib) - (bj|Ia)
+               IF (ABS(c_ss) > EPSILON(1.0_dp)) THEN
+                  bounds_2d(1, 1) = 1
+                  bounds_2d(2, 1) = nhomo
+                  bounds_2d(:, 2) = bounds_1d
+
+                  CALL dbcsr_t_contract(alpha=dbcsr_scalar(-1.0_dp), tensor_1=oI_Y(ido_mo), tensor_2=ja_X, &
+                                        beta=dbcsr_scalar(1.0_dp), tensor_3=aj_Ib, contract_1=[2], &
+                                        notcontract_1=[1], contract_2=[3], notcontract_2=[1, 2], &
+                                        map_1=[1], map_2=[2, 3], &
+                                        bounds_2=[nhomo + 1, nhomo + nlumo], bounds_3=bounds_2d)
+
+                  CALL calc_ss_ovv_contrib(parts(3), parts(4), aj_Ib, homo_evals, lumo_evals, omega_k, c_ss)
+
+               END IF ! c_ss not 0
+
+               CALL dbcsr_t_destroy(aj_Ib)
+            END DO
+
+            CALL mp_sum(parts, para_env%group)
+            s1 = parts(1); ds1 = parts(2)
+            s2 = parts(3); ds2 = parts(4)
+
+            !evaluate g and its derivative
+            g = eps_I - omega_k + s1 + s2
+            dg = -1.0_dp + ds1 + ds2
+
+            !compute the diff to the new step
+            diff = -g/dg
+
+            !and the new omega
+            omega_k = omega_k + diff
+            diff = diff*evolt
+
+            IF (output_unit > 0) THEN
+               WRITE (UNIT=output_unit, FMT="(T21,I18,F32.6)") &
```

Comment:
is it possible that this is a debug leftover? Plus, why always the `flushes`?

Reply:
It is not a debug leftover. For large systems, an iteration can be quite slow. This helps keeping track of the calculation status. The `flush` is also there for that reason. If the output is buffered, then printing this info is much less informative if it comes all at once.

Reply:
ok, I see